hudi-flink/src/main/java/org/apache/hudi/sink/bootstrap/BootstrapFunction.java

```diff
@@ -197,10 +219,70 @@ private void loadRecords(String partitionPath, Collector<O> out) {
         }
       }
     }
+  }
 
-    long cost = System.currentTimeMillis() - start;
-    LOG.info("Task [{}}:{}}] finish loading the index under partition {} and sending them to downstream, time cost: {} milliseconds.",
-        this.getClass().getSimpleName(), taskID, partitionPath, cost);
+  @SuppressWarnings("unchecked")
+  private void loadLogFiles(String partitionPath, Collector<O> out) throws Exception {
+    Option<HoodieInstant> latestCommitTime = this.hoodieTable.getMetaClient().getCommitsTimeline()
+            .filterCompletedInstants().lastInstant();
+    if (!latestCommitTime.isPresent()) {
+      return;
+    }
+
+    List<HoodieLogFile> logFiles = this.hoodieTable.getSliceView()
+            .getLatestFileSlicesBeforeOrOn(partitionPath, latestCommitTime.get().getTimestamp(), true)
+            .map(fileSlice -> fileSlice.getLogFiles().collect(toList()))
```

Comment:
To simplify the logic, fetch the fileslice directly and send the parquet and log record keys, the file group id is solid, which is recorded by the fileslice.

Reply:
I think so, I will modify this logic.